src/main/java/ui/components/pickers/LabelPickerUILogic.java

```diff
@@ -8,6 +8,28 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+/**
+ * Handles the logic of the label picker UI.
+ *
+ * The crux of the logic is as follows:
```

Comment:
From this line onward, you are explaining a specific method, and hence should be given as the method comment, not the class comment.